### ISSUE TITLE
Entry in tx_realurl_urldata won't deleted after remove a hidden page

### ISSUE THREAD

**Issue by ayacoo** [dmitryd/typo3-realurl#329]
First of all: Thanks for your work. In this times its important to thanks for work. And now my problem in a worst case scenario like my customer does.

Setup:
TYPO3 7.6.12
realurl 2.1.5

Reproduce (all steps via context menu in page tree)
- Add Page "Test" (ID1)
- Link ID1 in a Text Element (all is fine)
=> Table tx_realurl_urldata get an entry for "/test" and ID 1
- Hide Page "Test" (ID1)
- Rename Page "Test_empty" (ID1)
- Add Page "Test" (ID2) (I dont know why :D)
- Change link in RTE Element from ID1 to ID2
=> Table tx_realurl_urldata get an entry for "/test" and ID 2
- Delete Page "Test_empty" (ID1)

http://url/test/ map to ID1 an throws an 404. If i delete cache entry in database for ID 1 all is fine.  I hope this report helps to reproduce it.

Prio? Of course low ;-)

**Comment by dmitryd** [comment]
Related to https://github.com/dmitryd/typo3-realurl/commit/cb3b2fe5ed725b2b3f6586443f99306ec1a8e0a2.

**Comment by dmitryd** [comment]
Thank you for reporting.

**Comment by ayacoo** [comment]
Thanks for your fix! The request reaches the correct page :-)

Only the old entry for the first page (Page ID 1) still exists in database and expires in one month. But probably that is so desired?

**Comment by dmitryd** [comment]
> Only the old entry for the first page (Page ID 1) still exists in database and expires in one month. But probably that is so desired?

Yes, this is for SEO. Suppose you had a page named "Toothbrushes" and than you adjust it to "Toothbrushes for everybody". You do not want people coming to /toothbrushes/ from Google to get 404s and go away. You want them redirected to to a new URL. You would also want Google to pick up a new URL. Realurl does a 301 redirect from old addresses to new ones. This serves good for both customers and Google.

I will not close the issue just now because I want to check it again properly as soon as I have time.

**Comment by ayacoo** [comment]
Thanks for your good explanation.
